src/swap/v2/router/swap-router.ts

```diff
@@ -0,0 +1,243 @@
+import algosdk, {
+  Algodv2,
+  ALGORAND_MIN_TX_FEE,
+  getApplicationAddress,
+  Transaction
+} from "algosdk";
+
+import {CONTRACT_VERSION} from "../../../contract/constants";
+import {AccountInformation} from "../../../util/account/accountTypes";
+import {ALGO_ASSET_ID} from "../../../util/asset/assetConstants";
+import {getAssetId, isAlgo} from "../../../util/asset/assetUtils";
+import {SupportedNetwork} from "../../../util/commonTypes";
+import SwapQuoteError, {SwapQuoteErrorType} from "../../../util/error/SwapQuoteError";
+import {applySlippageToAmount, hasTinymanApiErrorShape} from "../../../util/util";
+import {getValidatorAppID} from "../../../validator";
+import {SwapType} from "../../constants";
+import {FetchSwapRouteQuotesPayload, SwapRouterResponse, SwapRoute} from "../../types";
+import {
+  V2_SWAP_APP_CALL_ARG_ENCODED,
+  V2_SWAP_APP_CALL_SWAP_TYPE_ARGS_ENCODED,
+  V2_SWAP_ROUTER_APP_ARGS_ENCODED
+} from "../constants";
+import {SWAP_ROUTER_INNER_TXN_COUNT, TINYMAN_ANALYTICS_API_BASE_URLS} from "./constants";
+import {
+  getAssetInFromSwapRoute,
+  getAssetOutFromSwapRoute,
+  getSwapRouterAppID
+} from "./util";
+
+/**
+ * Generates txns that would opt in the Swap Router Application to the assets used in the swap router
```

Comment:
Do this make swap router app opt-in to given assets? I didn't completely understand

Reply:
Yes, exactly.